File: DSA/Python/01_array.py
arr = []

while True:
    print("Enter 0 to exit")
    print("Enter 1 to push an element")
    print("Enter 2 to pop an element")
    print("Enter 3 to insert an element at desired position")
    print("Enter 4 to delete an element at desired position")
    print("Enter 5 to swap element positions")
    print("Enter 6 to show the array")

    choice = int(input())

    if not choice:
        break
    elif choice == 1: # append
        arr.append(int(input('Enter the element to be pushed : ')))
    elif choice == 2: # pop
        print(f'popped item : {arr.pop()}')
    elif choice == 3: # insert
        element = int(input('Enter element to be inserted : '))
        pos = int(input('Enter position to insert element : '))
        if pos > len(arr)+1:
            print('invalid position')
        else:
            arr.insert(pos-1,element)
    elif choice == 4: # remove from a position
        pos = int(input('Enter position of element to be removed : '))
        if pos > len(arr):
            print('invalid position')
        else:
            arr.pop(pos-1)
    elif choice == 5: # swap
        positions = [int(i)-1 for i in input("Enter positions of elements to be swapped comma separated : ").split(',')]
        temp = arr[positions[0]]
        arr[positions[0]] = arr[positions[1]]
        arr[positions[1]] = temp
    elif choice == 6: # display
        print(arr)
    else:
        print('Wrong Input')

File: Programming/DSA/C/01_array.c
# include <stdio.h>
# include <conio.h>
# include <stdlib.h>

void main()
{
	int size;
	int input = -1;
	int i;
	int pos;
	int element;
	int pos2;
	
	printf("\nEnter size of array : ");
	scanf("%d",&size);
	
	int *arr = (int*)malloc(sizeof(int)*size);
	int rear = 0;
	
	while(input!=0)
	{
		printf("\n\nEnter 0 to exit");
		printf("\nEnter 1 to push an element");
		printf("\nEnter 2 to pop an element");
		printf("\nEnter 3 to insert an element at desired position");
		printf("\nEnter 4 to delete an element at desired position");
		printf("\nEnter 5 to swap elements");
		printf("\nEnter 6 to show the array");
		
		printf("\n\nEnter Your Choice : ");
		scanf("%d", &input);
		
		switch(input)
		{
			case 0: // Exit
				printf("Exit Successful");
				break;
			case 1: // Push
				if(rear>size-1)
					printf("\nArray Overflow");
				else
				{
					printf("\nEnter number to be inserted : ");
					scanf("%d",&arr[rear]);
					rear++;
				}
				break;
			case 2: // Pop
				if(rear==0)
					printf("\nArray Underflow");
				else
				{
					printf("\nValue popped is : %d",arr[rear-1]);
					rear--;
				}
				break;
			case 3 : // Insert in pos
				printf("\nEnter the number and its position to be inserted : ");
				scanf("%d%d",&element,&pos);
				
				if (rear>size-1)
					printf("\nArray Overflow");
				else if(rear<pos-1)
					printf("\nArray Underflow");
				else if(rear == pos-1)
					arr[rear++] = element;
				else
				{
					printf("%d rear",rear);
					printf("%d pos",pos-1);
					for(i = rear; i>pos-1;i--)
						arr[i] = arr[i-1];
					arr[pos-1]=element;
					rear++;
				}
				break;
			case 4: // Delete in pos
				printf("\nEnter the position to be deleted : ");
				scanf("%d",&pos);
				
				if(rear<=pos-1)
					printf("\nArray Underflow");
				else
				{
					printf("\nElement deleted : %d", arr[pos-1]);
					for(i = pos-1;i<rear;i++)
						arr[i] = arr[i+1];
					rear--;
				}
				break;
			case 5: // swap elements
				printf("Enter positions of elements to be swapped : ");
				scanf("%d%d",&pos,&pos2);
				
				if(pos2>=rear || pos2 < 0 || pos < 0|| pos >=rear)
					printf("Invalid Positions");
				else
				{
					element = arr[pos-1];
					arr[pos-1] = arr[pos2-1];
					arr[pos2-1] = element;	
				}
				break;
			case 6: // print
				if(rear == 0)
					printf("Array Empty");
				else
					for(i=0;i<rear;i++)
						printf("%d\t",arr[i]);
				break;
			default :
				printf("\nWrong Input");
		}
	}
}
